config/config.go

```diff
@@ -902,17 +906,19 @@ func escapeEnv(value string) string {
 }
 
 func loadConfig(config string) ([]byte, error) {
-	u, err := url.Parse(config)
-	if err != nil {
-		return nil, err
-	}
+	if fetchURLRe.MatchString(config) {
+		u, err := url.Parse(config)
+		if err != nil {
+			return nil, err
+		}
 
-	switch u.Scheme {
-	case "https", "http":
-		return fetchConfig(u)
-	default:
-		// If it isn't a https scheme, try it as a file.
+		switch u.Scheme {
+		case "https", "http":
+			return fetchConfig(u)
+		}
```

Comment:
maybe have a default case that errors instead of falling through to reading it as a file.
```suggestion
		default:
			return nil, fmt.Errorf("scheme %q not supported", u.Scheme)
		}
```